### ISSUE TITLE
ssh-rsa considered deprecated, any chance at supporting ssh-ed25519 or ecdsa-sha2-nistp256 ?

### ISSUE THREAD

**Issue by brokenjacobs** [bluegreen-labs/phenocam_installation_tool_v2#5]
Some background here:

2020-02-14 Openssh released version 8.2 which officially deprecated support for ssh-rsa keys. The result is that as time moves on, more and more ssh implementations will not accept these keys.

https://www.openssh.com/txt/release-8.2
```
Future deprecation notice
=========================

It is now possible[1] to perform chosen-prefix attacks against the
SHA-1 hash algorithm for less than USD$50K. For this reason, we will
be disabling the "ssh-rsa" public key signature algorithm that depends
on SHA-1 by default in a near-future release.

This algorithm is unfortunately still used widely despite the
existence of better alternatives, being the only remaining public key
signature algorithm specified by the original SSH RFCs.

The better alternatives include:

 * The RFC8332 RSA SHA-2 signature algorithms rsa-sha2-256/512. These
   algorithms have the advantage of using the same key type as
   "ssh-rsa" but use the safe SHA-2 hash algorithms. These have been
   supported since OpenSSH 7.2 and are already used by default if the
   client and server support them.

 * The ssh-ed25519 signature algorithm. It has been supported in
   OpenSSH since release 6.5.

 * The RFC5656 ECDSA algorithms: ecdsa-sha2-nistp256/384/521. These
   have been supported by OpenSSH since release 5.7.

To check whether a server is using the weak ssh-rsa public key
algorithm for host authentication, try to connect to it after
removing the ssh-rsa algorithm from ssh(1)'s allowed list:

    ssh -oHostKeyAlgorithms=-ssh-rsa user@host

If the host key verification fails and no other supported host key
types are available, the server software on that host should be
upgraded.

A future release of OpenSSH will enable UpdateHostKeys by default
to allow the client to automatically migrate to better algorithms.
Users may consider enabling this option manually.

[1] "SHA-1 is a Shambles: First Chosen-Prefix Collision on SHA-1 and
    Application to the PGP Web of Trust" Leurent, G and Peyrin, T
    (2020) https://eprint.iacr.org/2020/014.pdf

```

TLDR; Openssh won't work your ssh-rsa keys out of the box anymore unless you enable deprecated algorithms.

I don't know the version of openssh/dropbear on these cameras as I don't have one in hand yet, but it would preferable to use a non-deprecated algorithm moving forward to save pain later.

**Comment by brokenjacobs** [comment]
I should clarify, rsa itself is fine for ssh it's just the signing algorithms that use sha-1 that are deprecated. If ed25519 isn't supported and ecdsa not desired, you can also use ssh-sha2-256 or ssh-sha2-512 key types.

**Comment by khufkens** [comment]
The best I can do ecdsa 521

**Comment by khufkens** [comment]
https://github.com/bluegreen-labs/phenocam_installation_tool_v2/commit/9452e0945f4aa53cf14ab116e3593fd54c87b603
